src/hooks/useSelector.js

```diff
@@ -46,13 +52,15 @@ export function useSelector(selector) {
     contextSub
   ])
 
+  const memoizedSelector = useMemo(() => selector, deps)
+
   const latestSubscriptionCallbackError = useRef()
-  const latestSelector = useRef(selector)
+  const latestSelector = useRef(memoizedSelector)
 
   let selectedState = undefined
 
   try {
-    selectedState = latestSelector.current(store.getState())
+    selectedState = memoizedSelector(store.getState())
```

Comment:
using `latestSelector.current` was wrong, since it was an stale selector. This change fixes that issue, which makes the test for stale props succeed.

Reply:
Hah, nice catch :)